### PR TITLE
feature: build binary for OS and arch by default

### PR DESCRIPTION
**Pull request by TerryHowe**
**What this PR does / why we need it**:

Build the binary by default for the current OS and arch.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,6 +43,10 @@ endif
 LDFLAGS += -X $(PROJECT_PKG)/internal/version.GitCommit=${GIT_COMMIT}
 LDFLAGS += -X $(PROJECT_PKG)/internal/version.GitTreeState=${GIT_DIRTY}
 
+.PHONY: default
+default: test build-mac-$(ARCH)
```
Comment (qweeah) [inline Makefile:47]: Should it be
```suggestion
default: test build-$(OS)-$(ARCH)
```

**Review by qweeah — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -18,6 +18,18 @@ GIT_COMMIT  = $(shell git rev-parse HEAD)
...
+
+ifeq ($(OSNAME),Darwin)
+  OS = mac
+else
+ifeq ($(OSNAME),Linux)
+  OS = linux
+else
+  OS = windows
+endif
+endif
```
Comment (qweeah) [inline Makefile:32]: extra line?
```suggestion
```

File: Makefile [hunk 3]
```diff
@@ -18,6 +18,16 @@ GIT_COMMIT  = $(shell git rev-parse HEAD)
...
+OSNAME      = $(shell uname -o)
+ARCH        = $(shell uname -m)
+
+ifeq ($(OSNAME),Darwin)
+  OS = mac
+else ifeq ($(OSNAME),Linux)
+  OS = linux
+else
+  OS = windows
+endif
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply Makefile]: Thanks. Updated and my goal here is just to support amd64, arm64 on mac or Linux.

**Review by qweeah — APPROVED** [review]
LGTM. Validated on Ubuntu and MAC

**Pull request merged**